tests/tests_unit/test_cdf_tk/test_load.py

```diff
@@ -464,3 +444,75 @@ def test_deploy_resource_order(self, cognite_client_approval: ApprovalCogniteCli
         actual_order = [view["externalId"] for view in views]
 
         assert actual_order == expected_order
+
+
+def find_subclasses(cls):
+    subclasses = set()
+    for subclass in cls.__subclasses__():
+        subclasses |= find_subclasses(subclass)  # Recursive call to find indirect subclasses
+        if abc.ABC in subclass.__bases__:
+            continue
+        subclasses.add(subclass)
+    return subclasses
+
+
+class TestListDictConsistency:
+    @pytest.mark.parametrize("Loader", sorted(find_subclasses(ResourceLoader), key=lambda x: x.folder_name))
+    def test_fake_resource_generator(
+        self, Loader: type[ResourceLoader], cdf_tool_config: CDFToolConfig, monkeypatch: MonkeyPatch
+    ):
+        fakegenerator = FakeCogniteResourceGenerator(seed=1337)
+
+        loader = Loader.create_loader(cdf_tool_config)
+        instance = fakegenerator.create_instance(loader.resource_write_cls)
+
+        assert isinstance(instance, loader.resource_write_cls)
+
+    @pytest.mark.parametrize("Loader", sorted(find_subclasses(ResourceLoader), key=lambda x: x.folder_name))
+    def test_loader_takes_dict(
+        self, Loader: type[ResourceLoader], cdf_tool_config: CDFToolConfig, monkeypatch: MonkeyPatch
+    ):
+        fakegenerator = FakeCogniteResourceGenerator(seed=1337)
+
+        loader = Loader.create_loader(cdf_tool_config)
+        instance = fakegenerator.create_instance(loader.resource_cls)
+
+        # special cases
+        if isinstance(instance, TransformationSchedule):
+            del instance.id  # Client validation does not allow id and externalid to be set simultaneously
+        elif isinstance(instance, (Transformation, FileMetadata)):
+            pytest.skip("Skipped loaders that require secondary files")
+        elif isinstance(instance, (Edge, Node)):
+            pytest.skip(f"Skipping {type(instance)} because it has special properties")
+
+        mock_read_yaml_file({"dict.yaml": instance.dump()}, monkeypatch)
+
+        loaded = loader.load_resource(filepath=Path("dict.yaml"), ToolGlobals=cdf_tool_config, skip_validation=True)
+        assert isinstance(
+            loaded, (loader.resource_write_cls, loader.list_write_cls)
+        ), f"loaded must be an instance of {loader.list_write_cls} or {loader.resource_write_cls} but is {type(loaded)}"
+
+    @pytest.mark.parametrize("Loader", sorted(find_subclasses(ResourceLoader), key=lambda x: x.folder_name))
+    def test_loader_takes_list(
+        self, Loader: type[ResourceLoader], cdf_tool_config: CDFToolConfig, monkeypatch: MonkeyPatch
+    ):
+        fakegenerator = FakeCogniteResourceGenerator(seed=1337)
+
+        loader = Loader.create_loader(cdf_tool_config)
+        instances = fakegenerator.create_instances(loader.list_cls)

```

Comment:
I guess this should be the write class?
```suggestion
        instances = fakegenerator.create_instances(loader.list_write_cls)

```